internal/index/indexscanner/scanner_test.go

```diff
@@ -90,6 +94,58 @@ func TestReadPluginFile(t *testing.T) {
 	}
 }
 
+func TestReadReceiptFromFile(t *testing.T) {
+	type args struct {
+		receiptFilePath string
+	}
+	tests := []struct {
+		name       string
+		args       args
+		wantStatus index.ReceiptStatus
+	}{
+		{
+			name: "read receipt of plugin without status",
+			args: args{
+				receiptFilePath: filepath.Join(testdataPath(t), "testindex", "receipts", "foo.yaml"),
+			},
+			wantStatus: index.ReceiptStatus{},
+		},
+		{
+			name: "read receipt of plugin from default index with status",
+			args: args{
+				receiptFilePath: filepath.Join(testdataPath(t), "testindex", "receipts", "foo-explicit.yaml"),
+			},
+			wantStatus: index.ReceiptStatus{
+				Source: index.SourceIndex{
+					Name: constants.DefaultIndexName,
+				},
+			},
+		},
+		{
+			name: "read receipt of plugin from custom index",
+			args: args{
+				receiptFilePath: filepath.Join(testdataPath(t), "testindex", "receipts", "custom.yaml"),
+			},
+			wantStatus: index.ReceiptStatus{
+				Source: index.SourceIndex{
+					Name: "custom",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotReceipt, err := ReadReceiptFromFile(tt.args.receiptFilePath)
+			if err != nil {
+				t.Fatalf("ReadReceiptFromFile() error: %v", err)
+			}
+			if diff := cmp.Diff(gotReceipt.Status, tt.wantStatus); diff != "" {
```

Comment:
I think cmp.Diff is frequently used as cmp.Diff(want, got) –you can remember from git diff, like: (old,new). This will work regardless, but just a nit.